File: src/homework06/controller/VendingMachineController.java
package homework06.controller;

import java.util.List;
import java.util.Map;
import homework06.service.VendingMachineImpl;

public class VendingMachineController {
    private VendingMachineImpl vendingMachineService = new VendingMachineImpl();

    public void addProduct(String productType, Map<String, Object> productData) {
        if (vendingMachineService.getProductTypes().contains(productType)) {
            vendingMachineService.addProduct(productType, productData);
        }
    }

    public List<String> getPpriceList() {
        return vendingMachineService.getProductPriceList();
    }

    public List<String> getProductByMatchPattern(Map<String, Object> matchPattern) {
        return vendingMachineService.getProducts(matchPattern);
    }
}

File: src/homework07/lib/controller/vending_machine_controller.dart
import '../service/vending_machine_impl.dart';

class VendingMachineController {
  final VendingMachineImpl _vendingMachineService = VendingMachineImpl();

  addProduct(String productType, Map<String, Object> productData) {
    if (_vendingMachineService.getProductTypes().contains(productType)) {
      _vendingMachineService.addProduct(productType, productData);
    }
  }

  List<String> getPpriceList() {
    return _vendingMachineService.getProductPriceList();
  }

  List<String> getProductByMatchPattern(Map<String, Object> matchPattern) {
    return _vendingMachineService.getProducts(matchPattern);
  }
}
